### PR TITLE
Glt 545 relocate parent

### PR DESCRIPTION
**Pull request by djcooke**
Moved parentId field to SampleAdditionalInfo, as it only applies to detailed samples. This also enables creating a composite unique key that is required (or at least helpful) for sample alias generation.

Also removed unused lastUpdated field from Sample. This field did not exist in the database, and the data is available via SampleChangeLog already

### CODE REVIEW

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java [hunk 1]
```diff
@@ -104,15 +104,15 @@ public Long create(SampleDto sampleDto) throws IOException {
     authorizationManager.throwIfNotWritable(sample);
     User user = authorizationManager.getCurrentUser();
 
-    if (sampleDto.getParentId() == null && isParentedSample(sample)) {
+    if (sampleDto.getSampleAdditionalInfo().getParentId() == null && isParentedSample(sample)) {
```
Comment (tonydebat) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: Check that sampleDto.getSampleAdditionalInfo() isn't null as well.

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java [hunk 2]
```diff
@@ -217,10 +216,10 @@ Sample to(SampleDto sampleDto) throws IOException {
     ServiceUtils.throwIfNull(project, "projectId", sampleDto.getProjectId());
     sample.setProject(project);
 
-    if (sampleDto.getParentId() != null) {
-      Sample parent = sampleDao.getSample(sampleDto.getParentId());
-      ServiceUtils.throwIfNull(parent, "parentId", sampleDto.getParentId());
-      sample.setParent(parent);
+    if (sampleDto.getSampleAdditionalInfo().getParentId() != null) {
```
Comment (tonydebat) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: Check that sampleDto.getSampleAdditionalInfo() isn't null as well.

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java [hunk 3]
```diff
@@ -80,9 +80,9 @@ public SampleDto getSample(@PathVariable("id") Long id, UriComponentsBuilder uri
   private static SampleDto writeUrls(SampleDto sampleDto, UriComponentsBuilder uriBuilder) {
     URI baseUri = uriBuilder.build().toUri();
     sampleDto.setUrl(UriComponentsBuilder.fromUri(baseUri).path("/rest/tree/sample/{id}").buildAndExpand(sampleDto.getId()).toUriString());
-    if (sampleDto.getParentId() != null) {
-      sampleDto.setParentUrl(
-          UriComponentsBuilder.fromUri(baseUri).path("/rest/tree/sample/{id}").buildAndExpand(sampleDto.getParentId()).toUriString());
+    if (sampleDto.getSampleAdditionalInfo().getParentId() != null) {
```
Comment (tonydebat) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java]: Check that sampleDto.getSampleAdditionalInfo() isn't null as well.

**Comment by tonydebat** [comment]
:+1:

**Comment by apmasell** [comment]
:+1:

**Pull request merged**